src/battle_main.c

```diff
@@ -4096,7 +4096,8 @@ static void HandleTurnActionSelectionState(void)
 
             // Do AI score computations here so we can use them in AI_TrySwitchOrUseItem
             if ((gBattleTypeFlags & BATTLE_TYPE_HAS_AI || IsWildMonSmart())
-                && (IsBattlerAIControlled(gActiveBattler) && !(gBattleTypeFlags & BATTLE_TYPE_PALACE))) {
+                    && (BattlerHasAi(gActiveBattler) && !(gBattleTypeFlags & BATTLE_TYPE_PALACE)))
```

Comment:
The suggestion was to stack the conditions  so the 2nd line's checks could be read right below the `if` statement and the checks in its same line. You only aded even more spacing which is not what I suggested 😂 ...

```c
            if ((gBattleTypeFlags & BATTLE_TYPE_HAS_AI || IsWildMonSmart())
             && (BattlerHasAi(gActiveBattler) && !(gBattleTypeFlags & BATTLE_TYPE_PALACE)))
```

Reply:
As per discord discussion it doesn't seem like there is a universal formatting for things like this so I would prefer to leave it as is